lib/icinga/apiactions.cpp

```diff
@@ -555,5 +555,186 @@ Dictionary::Ptr ApiActions::GenerateTicket(const ConfigObject::Ptr&,
+    Checkable::Ptr checkable = static_pointer_cast<Checkable>(object);
+    if (! checkable)
+        return ApiActions::CreateResult(404, "Can't schedule a command execution for non-existent object.");
+
+    ObjectLock oLock (checkable);
+
+    Host::Ptr host;
+    Service::Ptr service;
+    tie(host, service) = GetHostService(checkable);
+
+    String endpoint = "$command_endpoint$";
+    if (params->Contains("endpoint"))
+        endpoint = params->Get("endpoint");
+
+    /* Resolve endpoint macro */
+    MacroProcessor::ResolverList resolvers;
+    if (service)
+        resolvers.emplace_back("service", service);
+    resolvers.emplace_back("host", host);
+    resolvers.emplace_back("icinga", IcingaApplication::GetInstance());
+
+    Dictionary::Ptr resolvedMacros;
+    bool useResolvedMacros;
+
+    if (params->Contains("macros")) {
+        resolvedMacros = HttpUtility::GetLastParameter(params, "macros");
+        useResolvedMacros = true;
+    } else {
+        resolvedMacros = new Dictionary();
+        useResolvedMacros = false;
+    }
+
+    String resolved_endpoint = MacroProcessor::ResolveMacros(endpoint,
+                                                             resolvers,
+                                                             checkable->GetLastCheckResult(),
+                                                             nullptr,
+                                                             MacroProcessor::EscapeCallback(),
+                                                             resolvedMacros,
+                                                             useResolvedMacros);
+
+    /* Check if resolved_endpoint is not empty */
+    if (resolved_endpoint.IsEmpty())
+        return ApiActions::CreateResult(400, "Endpoint must not be empty.");
+
+    /* Check if endpoint exists */
+    if (!Endpoint::GetByName(resolved_endpoint))
+        return ApiActions::CreateResult(404, "Can't find a valid endpoint for '" + resolved_endpoint + "'.");
+
+    /* Get command_type */
+    String command_type = "EventCommand";
+    if (params->Contains("command_type"))
+        command_type = HttpUtility::GetLastParameter(params, "command_type");
+
+    /* Validate command_type */
+    if (command_type != "EventCommand" && command_type != "CheckCommand" && command_type != "NotificationCommand")
+        return ApiActions::CreateResult(400, "Invalid command_type '" + command_type + "'.");
+
+    /* Get command */
+    String command;
+    if (!params->Contains("command")) {
+        if (command_type == "CheckCommand" ) {
+            command = "$check_command$";
+        } else if (command_type == "EventCommand") {
+            command = "$event_command$";
+        } else if (command_type == "NotificationCommand") {
+            command = "$notification_command$";
+        } else {
+            return ApiActions::CreateResult(404, "Can't infer 'command' from command_type '" + command_type + "'");
```

Comment:
In which case you run into this block?

Reply:
You're right, this is an artifact from before the validation of the command_type, we'll remove it.